File: kattis/foundation/planina/planina.py
import sys
import time


class planina:

    debug = False

    @staticmethod
    def dPrint(*args, **kwargs):
        if planina.debug:
            print(*args, **kwargs)

    @staticmethod
    def isValidIterationNumber(value: int) -> bool:
        min = 1
        max = 15
        return min <= value <= max

    @staticmethod
    def generateOutput():
        # Take the line input
        inputIterationNumber = sys.stdin.readline().strip()
        if not input:
            planina.dPrint("Input error")
            return

        # Start the timer
        startRuntime = time.perf_counter()
 
        # Find valid input
        if not inputIterationNumber.isdigit():
            planina.dPrint("Input is not valid")
            return

        # Find iteration number
        iterationNumber = int(inputIterationNumber)
        if not planina.isValidIterationNumber(iterationNumber):
            planina.dPrint("Wrong iteration number")
            return

        # Find the number of points
        planina.dPrint(f"Totla Iteration Number: {iterationNumber}")
        pointsOnASingleArm = 2
        
        for i in range(iterationNumber):
            pointsOnASingleArm = ((pointsOnASingleArm - 1) * 2 ) + 1
            
            planina.dPrint(f"Current Iteration Number: {i+1}")
            if i+1 == iterationNumber:
                planina.dPrint("End of Iteration.")
                break

        # Find number of points
        planina.dPrint("Number of Points:")
        print(pow(pointsOnASingleArm, 2))

        # Stop the timer here
        endRuntime = time.perf_counter()
        totalRuntime = endRuntime - startRuntime

        planina.dPrint(f"[Algorithmic Runtime] {totalRuntime:.6f} seconds")


if __name__ == "__main__":
    planina.generateOutput()

File: kattis/foundation/planina/planina.java
package kattis.foundation.planina;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class planina {

    private static final boolean debug = false;

    public static void main(String[] args) {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        try {
            // Take the line input
            String inputIterationNumber = bufferedReader.readLine();
            if (inputIterationNumber == null || inputIterationNumber.trim().isEmpty()) {
                dPrint("Input error");
                return;
            }

            // Start the timer
            long startRuntime = System.nanoTime();

            // Find valid input
            if (!isDigit(inputIterationNumber)) {
                dPrint("Input is not valid");
                return;
            }

            // Find iteration number
            int iterationNumber = Integer.parseInt(inputIterationNumber);
            if (!isValidIterationNumber(iterationNumber)) {
                dPrint("Wrong iteration number");
                return;
            }

            // Find the number of points on a single side
            dPrint("Total Iteration Number: " + iterationNumber);

            // Rules: (2^N + 1)^2
            // Math.pow returns double, so we cast to int for points on a side
            int pointsOnSide = ((int) Math.pow(2, iterationNumber)) + 1;
            dPrint("Points on a single side: " + pointsOnSide);

            // Find the total number of points
            int numberOfPoints = pointsOnSide * pointsOnSide;
            dPrint("Number of Points:");
            System.out.println(numberOfPoints);

            // Stop the timer here
            long endRuntime = System.nanoTime();
            double totalRuntime = (endRuntime - startRuntime) / 1_000_000_000.0;

            dPrint(String.format("[Algorithmic Runtime] %.6f seconds", totalRuntime));

        } catch (IOException e) {
            dPrint("IOException occurred: " + e.getMessage());
        }
    }

    private static void dPrint(Object... args) {
        if (debug) {
            StringBuilder sb = new StringBuilder();
            for (Object arg : args) {
                sb.append(arg).append(" ");
            }
            System.out.println(sb.toString().trim());
        }
    }

    private static boolean isDigit(String str) {
        for (char c : str.toCharArray()) {
            if (!Character.isDigit(c)) {
                return false;
            }
        }
        return true;
    }

    private static boolean isValidIterationNumber(int value) {
        int min = 1;
        int max = 15;
        return value >= min && value <= max;
    }
}